lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -480,6 +480,10 @@ public function commit($entity = null)
         } catch (Throwable $e) {
             $this->em->close();
 
+            if ($this->em instanceof EntityManager) {
+                $this->em->setClosingThrowable($e);
+            }
+
```

Comment:
I'd honestly prefer this kind of logic:

```suggestion
            $this->em->close($e);
```

I'm aware that having an `EntityManagerInterface` may not allow us to do this, but it would be nicer than exposing a public setter to inject the closing throwable.

Reply:
If the parameter is optional, there won't be any violation: https://3v4l.org/6MsTY and SA tools will be fine with it: https://phpstan.org/r/b7f1a19d-b90d-423c-8a67-99f278f2a624

Reply:
Did we do such changes in a patch release in the past?

Reply:
Oh right, I forgot to mention the target branch. Let's retarget to 2.18.x

Reply:
Thanks for the clarification and the proof. It's something I wouldn't approve in our codebase that is why I was so hesitant. I will push the refactor, test on 2.18 and send it back.

Reply:
You will be able to feel better when making things right again on 3.0.x :)